Automattic/jetpack
# break social funcionality into a package

This PR breaks the Social plugin into a package and a plugin that's just a wrapper to the package.

It's a proof of concept of what should be the skeleton of any new Jetpack stand-alone plugin.

https://github.com/Automattic/jetpack/pull/23505 shows how it would be bundled into the Jetpack plugin.

#### Changes proposed in this Pull Request:

* Breaks the Social plugin into a package + plugin

#### Jetpack product discussion


p1HpG7-dFB-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Disable all jetpack plugins and leave only Social active
* Confirm Jetpack menu is present and Social is there
* Confirm you can connect Jetpack from the Social plugin
* Confirm My Jetpack shows up once the connection is established
* Confirm you can disconnect

Checkout https://github.com/Automattic/jetpack/pull/23505

This PR shows how a stand-alone plugin will be bundled into the Jetpack plugin.

Run the same tests above only with Jetpack enabled. See that Social menu looks the same.

Run the same tests with both plugins enabled. 

The experience must be the same in all cases.
